src/main/java/laddergame/domain/exception/BlankException.java

```diff
@@ -0,0 +1,10 @@
+package laddergame.domain.exception;
+
+public class BlankException extends IllegalArgumentException {
+
+    public static final String errorMessage = "[ERROR] 공백이 포함될 수 없습니다.";
+
+    public BlankException() {
+        super(errorMessage);
+    }
```

Comment:
예외는 어떤 오류인지 설명하는 것 만큼 어떤 상황에서 발생했는지 알리는 것도 중요합니다. 해당 예외의 경우 아래 처럼 처리할 수 있을거에요. (예시임) 
```suggestion
    public BlankException(String name) {
        super(errorMessage + " 입력된 유저 이름 : " + name);
    }
```

Reply:
✔️ 커스텀 예외의 범위 에 대한 답변을 여기서 진행하겠습니다!

1. 커스텀 예외가 범용적일 필요가 없습니다. 발생가능한 도메인 예외의 개수 만큼 존재해도 됩니다. ParticipantName을 parsing하다가 4가지 경우의 수로 예외가 발생할 수 있다면, 4개를 만들어도 됩니다. (하지만 아마 만들고나시면 좀 과하다 싶으셔서 줄이실 거에요 ㅎㅎ)
    적절한 중간 값을 찾아주세요.

    저희팀 같은 경우 예외 처리를 몇 가지 큰 범위의 커스텀 예외를 만들어 놓고(권한 에러, 비즈니스 에러, 통신 에러 등등) 그 중 범위에 해당하는 에러를 상속해서 만듭니다. 예외 메시지는 그냥 그때그때 주입하는 방식을 사용합니다 (에러가 발생되는 지점에서 만들어 넣기) 메시지의 원칙은 최대한 에러 상황을 잘 표현해주고, 어떤 값이 들어와서 오류가 발생했는지 알려주는 정도만 신경씁니다. 

2. 예외 메시지 테스트 같은 경우 "예외 메시지 변경" 이라는 세부 사항에 흔들리고 싶지 않다면 세분화된 커스텀 예외를 사용하면 됩니다. 예외 임계치에 해당하는 숫자가 변경되어도 현재도 잘 활용하고 계신 assertThatThrownBy().isInstanceOf()를 활용하시면 그대로 발생할 것이기 때문에, 예외 메시지가 아닌 타입을 통한 테스트가 가능해 질거에요. 사실 저는 그렇게 까진 안하고요, 이런 경우엔 예외 메시지를 상세하게 해주고 테스트에 하드 코딩을 하는 한이 있어도 한번 더 관리하는 걸 선택합니다. 테스트 관리의 피곤함 보다 좀더 자세한 예외 메시지를 선택했다고 봐주셔용

`테스트 코드의 예외 메시지 테스트를 쉽게 만든다 vs 도메인에서 관리를 조금 더 쉽게 한다`
요 질문을 보니 String.format()을 쓰냐 안쓰냐 에 대한 질문이었던 것 같네요 ㅎ

택 1 하라면 저는 후자입니다.(씁니다) 테스트도 중요하지만 궁극적으로 테스트는 프로덕션 코드의 보호가 목적이기 때문에 테스트가 쉬워지는 방향보다 항상 프로덕션 코드를 쉽게 만드는 걸 추구합니다.